language/evm/move-ethereum-abi/src/abi_signature_type.rs

```diff
@@ -30,3 +37,242 @@ pub struct ABIJsonSignature {
+        for input in &self.inputs {
+            let pt_name = input.name.clone();
+            let pt_result = input.convert_type_to_param_type();
+            if let Ok(pt) = pt_result {
+                let param = Param {
+                    name: pt_name,
+                    kind: pt,
+                    internal_type: None,
+                };
+                inputs.push(param);
+            } else {
+                return Err(anyhow!(ABI_PARSE_ERR_MSG));
+            }
+        }
+        Ok(Constructor { inputs })
+    }
+}
+
+fn convert_state_mutability(st_mut_str: &str) -> anyhow::Result<StateMutability> {
+    if st_mut_str == "payable" {
+        Ok(StateMutability::Payable)
+    } else if st_mut_str == "nonpayable" {
+        Ok(StateMutability::NonPayable)
+    } else if st_mut_str == "pure" {
+        Ok(StateMutability::Pure)
+    } else if st_mut_str == "view" {
+        Ok(StateMutability::View)
+    } else {
+        Err(anyhow!(ABI_PARSE_ERR_MSG))
+    }
+}
+
+fn check_type_int_range(num: usize) -> bool {
+    (8..=256).contains(&num) && num % 8 == 0
+}
+
+fn check_static_bytes_range(num: usize) -> bool {
+    (1..=32).contains(&num)
+}
+
+fn check_simple_type_prefix(ty_str: &str) -> bool {
+    /// Prefixes of value, bytes and string related types
+    const SIMPLE_TYPE_PREFIX: &[&str] = &[
+        "uint", "int", "ufixed", "fixed", "bool", "address", "bytes", "string",
+    ];
+    for prefix in SIMPLE_TYPE_PREFIX {
+        if ty_str.starts_with(prefix) {
+            return true;
+        }
+    }
+    false
+}
+
+fn convert_solidity_type(ty_str: &str) -> anyhow::Result<ParamType> {
+    let trimmed_ty_str = ty_str.trim();
+    if trimmed_ty_str.contains('[') {
+        // array type
+        convert_array(trimmed_ty_str)
+    } else if check_simple_type_prefix(trimmed_ty_str) {
+        // primitive and byte types
+        convert_simple_type(trimmed_ty_str)
+    } else {
+        Err(anyhow!(ABI_PARSE_ERR_MSG))
+    }
+}
+
+fn convert_array(ty_str: &str) -> anyhow::Result<ParamType> {
+    let last_pos = ty_str.rfind('[').context(ABI_PARSE_ERR_MSG)?;
+    let out_type = convert_solidity_type(&ty_str[..last_pos])?;
+    let last_indice_str = &ty_str[last_pos..].trim();
+    if last_indice_str.len() >= 2
+        && last_indice_str.starts_with('[')
+        && last_indice_str.ends_with(']')
+    {
+        let length_opt = last_indice_str[1..last_indice_str.len() - 1].trim();
+        if !length_opt.is_empty() {
+            let size = length_opt.parse::<usize>().context(ABI_PARSE_ERR_MSG)?;
+            if size == 0 {
+                return Err(anyhow!(ABI_PARSE_ERR_MSG_ZERO_SIZE));
+            }
+            return Ok(ParamType::FixedArray(Box::new(out_type), size));
+        } else {
+            return Ok(ParamType::Array(Box::new(out_type)));
+        }
+    }
+    Err(anyhow!(ABI_PARSE_ERR_MSG_ARRAY_TYPE))
+}
+
+fn convert_simple_type(ty_str: &str) -> anyhow::Result<ParamType> {
```

Comment:
Was wondering if we can use `https://docs.rs/ethabi/latest/ethabi/param_type/struct.Reader.html#method.read` to replace those parsing steps.